File: Programming/ecobike_rental_server/TKXDPM_API/Model/Transaction.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TKXDPM_API.Model
{
    public class Transaction
    {
        [Key] public int TransactionId { get; set; }
        public int RentalId { get; set; }
        public Rental Rental { get; set; }
        public PaymentStatus PaymentStatus { get; set; }
        public DateTime BookedStartDateTime { get; set; } = DateTime.MinValue;
        public DateTime BookedEndDateTime { get; set; } = DateTime.MinValue;
        public DateTime ActualStartDateTime { get; set; } = DateTime.MinValue;
        public DateTime ActualEndDateTime { get; set; } = DateTime.MinValue;
    }

    public class TransactionResponse
    {
        [Key] public int TransactionId { get; set; }
        public PaymentStatus PaymentStatus { get; set; }
        public string BookedStartDateTime { get; set; }
        public string BookedEndDateTime { get; set; }
        public string ActualStartDateTime { get; set; }
        public string ActualEndDateTime { get; set; }
    }

    public enum PaymentStatus
    {
        NotPay = 1,
        Deposit = 2,
        Paid = 3
    }
}

File: Programming/ecobike_rental/lib/helper/api/response/transaction.dart
class TransactionResponse {
  int transactionId;
  bool rental;
  bool paymentStatus;
  String bookedStartDateTime;
  String bookedEndDateTime;
  String actualStartDateTime;
  String actualEndDateTime;

  TransactionResponse(
      {this.transactionId,
      this.rental,
      this.paymentStatus,
      this.bookedStartDateTime,
      this.bookedEndDateTime,
      this.actualStartDateTime,
      this.actualEndDateTime});

  TransactionResponse.fromJson(Map<String, dynamic> json) {
    transactionId = json['transactionId'];
    rental = json['rental'];
    paymentStatus = json['paymentStatus'];
    bookedStartDateTime = json['bookedStartDateTime'];
    bookedEndDateTime = json['bookedEndDateTime'];
    actualStartDateTime = json['actualStartDateTime'];
    actualEndDateTime = json['actualEndDateTime'];
  }

  Map<String, dynamic> toJson() {
    final data = <String, dynamic>{};
    data['transactionId'] = transactionId;
    data['rental'] = rental;
    data['paymentStatus'] = paymentStatus;
    data['bookedStartDateTime'] = bookedStartDateTime;
    data['bookedEndDateTime'] = bookedEndDateTime;
    data['actualStartDateTime'] = actualStartDateTime;
    data['actualEndDateTime'] = actualEndDateTime;
    return data;
  }
}
